base_layer/core/src/types.rs

```diff
@@ -111,3 +118,47 @@ impl ByteArray for RangeProof {
         &self.0
     }
 }
+
+impl fmt::Display for RangeProof {
```

Comment:
You may as well move RangeProof to its own file, since these impls clearly don't go in types.rs